Added ability to add Named Entities

benpocalypse opened the pull request:
Added a feature to the ECS library so that when Entities are added, they can be given a unique String Id so that you can get specific Entities in the future, rather than just retrieving them by Component type.

I was using a different ECS library in my game that was much too slow, and replaced it with this excellent ECS system that is fast, but wanted to add a couple features that made it more of a drop-in replacement.

benpocalypse committed:
Added ability to add Named Entities

Added a feature to the ECS library so that when Entities are added, they can be given a unique String Id so that you can get specific Entities in the future, rather than just retrieving them by Component type.

benpocalypse commented:
Feel free to reject this PR if you have no interest in adding it to the ECS library, I just thought it was useful.

adizhavo commented:
Hi,

Thank you for your contribution. This was actually smth i was thinking some time ago but felt a bit lazy to update the library. I think that this is a very useful PR.

It would be nice to add some unit tests for this feature for both the Entity and EntityMatcher and maybe update the README file before the merge.

I will be also setting up a CI agent for builds and test cases.

benpocalypse commented:
Great!

I can add some Unit Tests today...something everyone should do. Don't accept the PR until then.

benpocalypse committed:
Added a few Unit Tests

As per the original authors request, I have added unit tests to check the functionality of the new features I added to the ECS.
